File: src/main/kotlin/com/phodal/gradal/plugins/gradle/facet/GradleFacetType.java
package com.phodal.gradal.plugins.gradle.facet;

import com.intellij.facet.Facet;
import com.intellij.facet.FacetType;
import com.intellij.openapi.module.Module;
import com.intellij.openapi.module.ModuleType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class GradleFacetType extends FacetType<GradleFacet, GradleFacetConfiguration> {
  public GradleFacetType() {
    super(GradleFacet.getFacetTypeId(), GradleFacet.getFacetId(), GradleFacet.getFacetName());
  }

  @NotNull
  @Override
  public GradleFacetConfiguration createDefaultConfiguration() {
    return new GradleFacetConfiguration();
  }

  @NotNull
  @Override
  public GradleFacet createFacet(@NotNull Module module,
                                 @NotNull String name,
                                 @NotNull GradleFacetConfiguration configuration,
                                 @Nullable Facet underlyingFacet) {
    return new GradleFacet(module, name, configuration);
  }

  @Override
  public boolean isSuitableModuleType(ModuleType moduleType) {
    return true;
  }
}

File: src/main/kotlin/com/phodal/gradal/plugins/gradle/facet/GradleFacetType.kt
package com.phodal.gradal.plugins.gradle.facet

import com.intellij.facet.Facet
import com.intellij.facet.FacetType
import com.intellij.openapi.module.Module
import com.intellij.openapi.module.ModuleType

class GradleFacetType : FacetType<GradleFacet, GradleFacetConfiguration>(GradleFacet.facetTypeId, GradleFacet.facetId, GradleFacet.facetName) {
    override fun createDefaultConfiguration(): GradleFacetConfiguration {
        return GradleFacetConfiguration()
    }

    override fun createFacet(module: Module,
                             name: String,
                             configuration: GradleFacetConfiguration,
                             underlyingFacet: Facet<*>?): GradleFacet {
        return GradleFacet(module, name, configuration)
    }

    override fun isSuitableModuleType(moduleType: ModuleType<*>?): Boolean {
        return true
    }
}
